src/v/coproc/event_listener.cc

```diff
@@ -45,108 +47,53 @@ kafka::client::client make_client() {
-            /// arrived in separate batches.
-            if (found == _active_ids.end()) {
-                enables.emplace_back(enable_copros_request::data{
-                  .id = id, .source_code = std::move(source)});
-            }
-        }
-    }
-    /// TODO: In the future, maybe it would be cleaner to have a add/remove
-    /// endpoint, instead of two seperate RPC endpoints
-    if (!enables.empty()) {
-        enable_copros_request req{.inputs = std::move(enables)};
-        auto resp = co_await _dispatcher.enable_coprocessors(std::move(req));
-        if (resp.has_error()) {
-            vlog(
-              coproclog.error,
-              "Failed to register coprocessors with the wasm engine: {}",
-              resp.error());
-            _offset = last_offset;
-            co_return;
-        } else {
-            for (script_id id : resp.value()) {
-                vlog(
-                  coproclog.info,
-                  "Successfully registered script with id: {}",
-                  id);
-                _active_ids.insert(id);
-            }
-        }
-    }
-    if (!disables.empty()) {
-        disable_copros_request req{.ids = std::move(disables)};
-        auto resp = co_await _dispatcher.disable_coprocessors(std::move(req));
-        if (resp.has_error()) {
-            vlog(
-              coproclog.error,
-              "Failed to make disable request to the wasm engine: {}",
-              resp.error());
-            /// In this case the code will follow a path that re-enters this
-            /// method with the same inputs, and the call to enable_copros above
-            /// succeeded but the call to disable_coprocessors failed, double
-            /// registrations will be avoided because the ids have entered the
-            /// \ref active_ids cache and will not be queued for re-registration
-            _offset = last_offset;
-        } else {
-            for (script_id id : resp.value()) {
-                _active_ids.erase(id);
-            }
-        }
-    }
+    co_await ss::parallel_for_each(
+      _handlers.begin(),
+      _handlers.end(),
+      [](std::pair<const event_type, wasm::event_handler*>& type_and_handler) {
+          return type_and_handler.second->stop();
+      });
+    co_await _gate.close().then([this] { return _client.stop(); });
 }
 
-event_listener::event_listener(ss::sharded<pacemaker>& pacemaker)
-  : _client(make_client())
-  , _dispatcher(pacemaker, _abort_source) {}
+event_listener::event_listener()
+  : _client(make_client()) {}
 
 ss::future<> event_listener::start() {
     (void)ss::with_gate(_gate, [this] {
-        return ss::do_until(
-          [this] { return _abort_source.abort_requested(); },
-          [this] {
-              return do_start().then([this] {
-                  return ss::sleep_abortable(1s, _abort_source)
-                    .handle_exception_type([](const ss::sleep_aborted&) {});
-              });
+        return ss::parallel_for_each(
```

Comment:
I assume that we want to guarantee that method `stop` can stop the event listener if it's called after the `start` method. But here, if we will call `stop` after `start` we might not stop anything because handlers might not be started yet. 

the first part of the async thing should be called via `co_await` and then the async thing can be started

Reply:
done